compiler/test/__snapshots__/arrays.0f9e7d37.0.snapshot

```diff
@@ -112,21 +118,21 @@ arrays › array_access
     (i32.add
      (i32.shl
       (if (result i32)
-       (i32.lt_u
-        (local.get $1)
+       (i32.lt_s
```

Comment:
Why did this change from `lt_u` to `lt_s`?

Reply:
Optimizations are a bit of a mystery. This is going to be a case of Binaryen having decided that this makes more sense now vs before 🤷 

FWIW, `lt_u` of anything and 0 is always going to be false so I'm not really sure what Binaryen was doing before. I saw today actually that there's an optimization that looks at your locals and may switch between signed and unsigned instructions if it thinks it makes sense. Really wild.